src/hdlc.rs

```diff
@@ -0,0 +1,53 @@
+use alloc::borrow::{Cow, ToOwned};
+use hdlcparse::{
+  llc::{parse_llc, Llc},
+  type3::HdlcFrame,
+};
+
+use crate::{DlmsDataLinkLayer, Error};
+
+#[derive(Debug)]
+pub enum HdlcDataLinkLayer {}
+
+fn validate_llc(llc: Llc) -> bool {
+  llc.dest_sap == 0xE6 && (llc.src_sap == 0xE6 || llc.src_sap == 0xE7) && llc.control == 0x00
```

Comment:
From https://www.dlms.com/files/Green_Book_Edition_9-Excerpt.pdf:

> - The value of the Destination_LSAP is 0xE6;
> - The value of the Source_LSAP is 0xE6 or 0xE7. The least significant bit is used as a command/response identifier. When set to 0, it identifies a ‘command’ and when set to 1 it identifies a “response”;

This seems similar to `control_information::Direction`. Command *to* server is `0xE6` and response *from* server is `0xE7`.

Reply:
Also Destination LSAP `0xFF` (broadcast) is possible it seems.

Reply:
Maybe it would make sense to move the `llc` feature from `hdlcparse` into this crate and make the `Llc` struct use more rich types, i.e.

```rust
enum Destination {
  Unicast,
  Broadcast,
}

enum MessageType {
  Command,
  Response,
}

struct Llc {
  destination: Destination,
  message_type: MessageType,
  quality: u8,
  information: u8,
}
```

Seems like there should also be an `information` field.

Reply:
I have added the struct back